src/reporters/search/reporter.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Ux } from '@salesforce/sf-plugins-core';
+import { SearchResult } from '@jsforce/jsforce-node';
+
+export abstract class SearchReporter {
+  public types: string[];
```

Comment:
this iterates the records to collect unique types, then each reporter iterates the records again to filter one type at a time.

Why not have a Map of <type, Record[]>.  Then you already have them organized (and once we can use Map.groupBy it'll be even cleaner!)